### PR TITLE
Remove unused stateless for VkPresentRegionsKHR

### PR DESCRIPTION
**Pull request by spencer-lunarg**
There were 2 `ValidateArray` that were not correct, this removes them and adds tests

### CODE REVIEW

File: layers/stateless/sl_wsi.cpp [hunk 1]
```diff
@@ -186,14 +186,6 @@ bool StatelessValidation::manual_PreCallValidateQueuePresentKHR(VkQueue queue, c
             skip |= ValidateStructPnext("QueuePresentKHR", "pCreateInfo->pNext->pNext", NULL, present_regions->pNext, 0, NULL,
                                         GeneratedVulkanHeaderVersion, "VUID-VkPresentInfoKHR-pNext-pNext",
                                         "VUID-VkPresentInfoKHR-sType-unique");
-            skip |= ValidateArray("QueuePresentKHR", "pCreateInfo->pNext->swapchainCount", "pCreateInfo->pNext->pRegions",
```
Comment (spencer-lunarg) [inline layers/stateless/sl_wsi.cpp:189]: already being done in generated code, tests hits this, makes sense why this had no VUID

File: layers/stateless/sl_wsi.cpp [hunk 2]
```diff
@@ -186,14 +186,6 @@ bool StatelessValidation::manual_PreCallValidateQueuePresentKHR(VkQueue queue, c
...
-                                  present_regions->swapchainCount, &present_regions->pRegions, true, false, kVUIDUndefined,
-                                  kVUIDUndefined);
-            for (uint32_t i = 0; i < present_regions->swapchainCount; ++i) {
-                skip |= ValidateArray("QueuePresentKHR", "pCreateInfo->pNext->pRegions[].rectangleCount",
```
Comment (spencer-lunarg) [inline layers/stateless/sl_wsi.cpp:193]: Spec:

> rectangleCount is the number of rectangles in pRectangles, or zero if the entire image has changed and should be presented.

This was checking if `rectangleCount` was zero, which is produced a false positive

File: layers/stateless/sl_cmd_buffer.cpp [hunk 3]
```diff
@@ -227,7 +227,7 @@ bool StatelessValidation::ValidateCmdBindVertexBuffers2(VkCommandBuffer commandB
                     not_null_msg = "pSizes is not NULL";
                 else
                     not_null_msg = "pStrides is not NULL";
-                const char *vuid_breach_msg = "%s: %s, so bindingCount must be greater that 0.";
+                const char *vuid_breach_msg = "%s: %s, so bindingCount must be greater than 0.";
                 skip |= LogError(commandBuffer, "VUID-vkCmdBindVertexBuffers2-bindingCount-arraylength", vuid_breach_msg, api_call,
```
Comment (juan-lunarg) [inline layers/stateless/sl_cmd_buffer.cpp:231]: Don't see much point to the `vuid_breach_msg` variable.

```suggestion
                skip |= LogError(commandBuffer, "VUID-vkCmdBindVertexBuffers2-bindingCount-arraylength", "%s: %s, so bindingCount must be greater than 0.", api_call,
```

**Review by juan-lunarg — APPROVED** [review]

File: layers/stateless/sl_cmd_buffer.cpp (hunk 3 above)
Comment (spencer-lunarg) [reply layers/stateless/sl_cmd_buffer.cpp:231]: agree, let me fix this while here

Comment (spencer-lunarg) [reply layers/stateless/sl_cmd_buffer.cpp:231]: I only found this because I mis-spelled `than` and noticed this code lol

**Pull request merged**